var/spack/repos/builtin/packages/ginkgo/package.py

```diff
@@ -57,10 +58,11 @@ class Ginkgo(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("cmake@3.13:", type="build", when="@1.4.0:1.6.0")
     depends_on("cmake@3.16:", type="build", when="@1.7.0:")
     depends_on("cmake@3.18:", type="build", when="+cuda@1.7.0:")
+    depends_on("cmake@3.21:", type="build", when="+rocm@1.8.0:")
     depends_on("cuda@9:", when="+cuda @:1.4.0")
     depends_on("cuda@9.2:", when="+cuda @1.5.0:")
     depends_on("cuda@10.1:", when="+cuda @1.7.0:")
-    depends_on("mpi", when="+mpi")
+    depends_on("mpi@3.1:", when="+mpi")
```

Comment:
I'm seeing a strange issue with this change. An old version of openmpi is getting picked up. I'm not sure I completely understand this.

```console
svcpetsc@petsc-gpu-02:/scratch/svcpetsc/spack$ ./bin/spack spec ginkgo ^openmpi |grep openmpi@
 -       ^openmpi@1.7.4%gcc@11.4.0~atomics~cuda~cxx~cxx_exceptions~gpfs~internal-hwloc~internal-libevent~internal-pmix~java~lustre~memchecker~openshmem~orterunprefix~romio+rsh~singularity~sqlite3~static~thread_multiple+vt+wrapper-rpath build_system=autotools fabrics=none romio-filesystem=none schedulers=none arch=linux-ubuntu22.04-zen4
```
However - if I revert this change - the correct/latest version gets used
```console
svcpetsc@petsc-gpu-02:/scratch/svcpetsc/spack$ git diff
diff --git a/var/spack/repos/builtin/packages/ginkgo/package.py b/var/spack/repos/builtin/packages/ginkgo/package.py
index b4638d9666..0c1f6a5089 100644
--- a/var/spack/repos/builtin/packages/ginkgo/package.py
+++ b/var/spack/repos/builtin/packages/ginkgo/package.py
@@ -62,7 +62,7 @@ class Ginkgo(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("cuda@9:", when="+cuda @:1.4.0")
     depends_on("cuda@9.2:", when="+cuda @1.5.0:")
     depends_on("cuda@10.1:", when="+cuda @1.7.0:")
-    depends_on("mpi@3.1:", when="+mpi")
+    depends_on("mpi", when="+mpi")
 
     depends_on("rocthrust", when="+rocm")
     depends_on("hipsparse", when="+rocm")
svcpetsc@petsc-gpu-02:/scratch/svcpetsc/spack$ ./bin/spack spec ginkgo ^openmpi |grep openmpi@
 -       ^openmpi@5.0.5%gcc@11.4.0~atomics~cuda~gpfs~internal-hwloc~internal-libevent~internal-pmix~java~lustre~memchecker~openshmem~romio+rsh~static+vt+wrapper-rpath build_system=autotools fabrics=none romio-filesystem=none schedulers=none arch=linux-ubuntu22.04-zen4
```

Reply:
@alalazo Do you know why this might be happening?

Reply:
Don't know the exact mpi version to openmpi version map,  but changing the format to the one used in mpich gets this working.
```console
svcpetsc@petsc-gpu-02:/scratch/svcpetsc/spack$ git diff
diff --git a/var/spack/repos/builtin/packages/openmpi/package.py b/var/spack/repos/builtin/packages/openmpi/package.py
index 22bc36c360..8560519c93 100644
--- a/var/spack/repos/builtin/packages/openmpi/package.py
+++ b/var/spack/repos/builtin/packages/openmpi/package.py
@@ -570,10 +570,10 @@ class Openmpi(AutotoolsPackage, CudaPackage):
     variant("internal-libevent", default=False, description="Use internal libevent")
     variant("openshmem", default=False, description="Enable building OpenSHMEM")
 
-    provides("mpi")
-    provides("mpi@:2.2", when="@1.6.5")
-    provides("mpi@:3.0", when="@1.7.5:")
-    provides("mpi@:3.1", when="@2.0.0:")
+    provides("mpi@:4.0")
+    provides("mpi@:3.1", when="@:4.1")
+    provides("mpi@:3.0", when="@:1.10")
+    provides("mpi@:2.2", when="@:1.7")
 
     if sys.platform != "darwin":
         depends_on("numactl")
svcpetsc@petsc-gpu-02:/scratch/svcpetsc/spack$ ./bin/spack spec ginkgo ^openmpi |grep openmpi@
 -       ^openmpi@5.0.5%gcc@11.4.0~atomics~cuda~gpfs~internal-hwloc~internal-libevent~internal-pmix~java~lustre~memchecker~openshmem~romio+rsh~static+vt+wrapper-rpath build_system=autotools fabrics=none romio-filesystem=none schedulers=none arch=linux-ubuntu22.04-x86_64
svcpetsc@petsc-gpu-02:/scratch/svcpetsc/spack$ 
```

Reply:
Updated fix at https://github.com/spack/spack/pull/46102

Reply:
Since #46102 is having issues - hoping  we can use https://github.com/spack/spack/pull/46292 until these issues are resolved.